src/framework/uicomponents/view/abstracttoolbarmodel.cpp

```diff
@@ -125,7 +125,26 @@ void AbstractToolBarModel::setItems(const ToolBarItemList& items)
     TRACEFUNC;
 
     beginResetModel();
-    m_items = items;
+
+    m_items.clear();
+
+    //! NOTE: make sure that we don't have two separators sequentially
+    bool isPreviousSeparator = false;
+
+    for (ToolBarItem* item : items) {
+        if (item->type() == ToolBarItemType::SEPARATOR) {
+            if (isPreviousSeparator) {
+                item->setVisible(false);
```

Comment:
Maybe it's better to just delete it and not add it?

Reply:
I wasn't sure what users of this class expected from this method (this method says that it would just set the items) and decided to hide

Reply:
ok, agree, I'll fix it